src/main/java/net/library/controller/UserController.java

```diff
@@ -17,15 +23,20 @@
 @RequestMapping(USERS)
 public class UserController {
 
+    @Autowired
     private final UserService service;
 
     public UserController(final UserService service) {
         this.service = service;
     }
 
     @GetMapping
-    public List<UserDto> getAllUsers() {
-        return UserMapper.toDto(service.getAllUsers());
+    public ResponseEntity<List<UserDto>> getAllUsersByFilter(
```

Comment:
You should return something like page too, because you use pagination and clients need to have info about size and offset